ibis/backends/snowflake/registry.py

```diff
@@ -159,7 +153,19 @@ def _extract_url_query(t, op):
             )
         ),
         # snowflake typeof only accepts VARIANT
-        ops.TypeOf: unary(lambda arg: sa.func.typeof(sa.cast(arg, VARIANT))),
+        ops.TypeOf: unary(lambda arg: sa.func.typeof(sa.func.to_variant(arg))),
+        ops.All: reduction(sa.func.booland_agg),
+        ops.NotAll: reduction(lambda arg: ~sa.func.booland_agg(arg)),
+        ops.Any: reduction(sa.func.boolor_agg),
+        ops.NotAny: reduction(lambda arg: ~sa.func.boolor_agg(arg)),
+        ops.BitAnd: reduction(sa.func.bitand_agg),
+        ops.BitOr: reduction(sa.func.bitor_agg),
+        ops.BitXor: reduction(sa.func.bitxor_agg),
+        ops.DateFromYMD: fixed_arity(sa.func.date_from_parts, 3),
+        ops.StringToTimestamp: fixed_arity(sa.func.to_timestamp_tz, 2),
+        ops.RegexExtract: fixed_arity(sa.func.regexp_substr, 3),
+        ops.RegexSearch: fixed_arity(sa.func.regexp_like, 2),
```

Comment:
Alternatively, we can use the REGEX operator, but I don't care.
```suggestion
        ops.RegexSearch: fixed_arity(lambda left, right: left.op('REGEXP')(right), 2),
```